server/src/main/java/org/opensearch/cluster/metadata/TemplatesMetadata.java

```diff
@@ -140,9 +140,9 @@ public static TemplatesMetadata fromXContent(XContentParser parser) throws IOExc
                 currentFieldName = parser.currentName();
             }
             if (currentFieldName != null) {
-                while ((token = parser.nextToken()) != XContentParser.Token.END_OBJECT) {
+                do {
                     builder.put(IndexTemplateMetadata.Builder.fromXContent(parser, parser.currentName()));
-                }
+                } while ((token = parser.nextToken()) != XContentParser.Token.END_OBJECT);
```

Comment:
Curious would we only miss "order" or any one entry

Reply:
I think toXContent always writes order (as a primitive int) and that's why the issue may happen only for the order field.

Reply:
https://github.com/opensearch-project/OpenSearch/blob/0a10aca350c6fdcf0861f372683a2ee1f8f4b783/server/src/main/java/org/opensearch/cluster/metadata/IndexTemplateMetadata.java#L413